lib/phoenix_component/declarative.ex

```diff
@@ -1040,7 +1057,7 @@ defmodule Phoenix.Component.Declarative do
       end)
 
     for {name, {line, _column, _type_value}} <- attrs,
-        not (has_global? and __global__?(caller_module, Atom.to_string(name))) do
+        not (has_global? and __global__?(caller_module, Atom.to_string(name), attrs_defs)) do
```

Comment:
Instead of `has_global?` (boolean), make it be the actual global attribute (or nil), and then pass the actual attribute includes to `__global__?` instead of doing a traversal based on the name there. :)